src/lib.rs

```diff
@@ -214,6 +217,50 @@ impl Client {
         self.get_multi(keys).await
     }
 
+    // Gets the given key with additional metadata.
+    ///
+    /// If the key is found, `Some(Value)` is returned, describing the metadata and data of the key.
+    ///
+    /// Otherwise, `None` is returned.
+    ///
+    /// Supported meta flags:
+    /// - h: return whether item has been hit before as a 0 or 1
+    /// - l: return time since item was last accessed in seconds
+    /// - t: return item TTL remaining in seconds (-1 for unlimited)
+    pub async fn meta_get<K: AsRef<[u8]>>(
+        &mut self,
+        key: K,
+        meta_flags: Option<&[&str]>,
+    ) -> Result<Option<Value>, Error> {
+        self.conn.write_all(b"mg ").await?;
+        self.conn
+            .write_all(Self::validate_key_length(key.as_ref())?)
+            .await?;
+        self.conn.write_all(b" ").await?;
+        if let Some(flags) = meta_flags {
+            self.conn.write_all(flags.join(" ").as_bytes()).await?;
+        }
+        self.conn.write_all(b"\r\n").await?;
+        self.conn.flush().await?;
+
+        match self.drive_receive(parse_meta_get_response).await? {
+            Response::Status(Status::NotFound) => Ok(None),
+            Response::Status(s) => Err(s.into()),
+            Response::Data(d) => d
+                .map(|mut items| {
+                    if items.len() != 1 {
+                        Err(Status::Error(ErrorKind::Protocol(None)).into())
```

Comment:
Do we have a more verbose error that we can surface here? What is this error case?

Reply:
I think this can actually just drop out.  It was carried over from the ASCII implementation but it doesn't actually make sense in this context.

In the ASCII protocol, theoretically if there is no key validation for whitespace in the client layer that uses this crate, it's possible to construct a `get` operation with either a key split by whitespace or multiple keys, which could yield > 1 result when the `get` API is meant to either fetch a single result or nothing.  In the ASCII protocol, `get_multi` is just a `get` with many keys, and the memcached server understands that syntax.  This is not the case in the meta protocol - `mg` can only be pipelined, so there can only ever be one result or `None`.